uefi-firmware.nix

```diff
@@ -127,10 +136,19 @@ let
         chmod -R u+w BaseTools
       '';
 
+      inherit publicCertificateDer requiredSystemFeatures;
```

Comment:
This only requires the public key, which can be added to the nix store. So t his derivation doesn't need `requiredSystemFeatures`

Reply:
Also, why does `publicCertificateDer` get inherited here? It'd end up as an environment variable inside the build.

Reply:
I had thought about this for awhile. While I agree that the public key can freely be added to the nix store, the option `publicCertificateDer` could be represented as `"/path/to/pubkey"` instead of say `./path/to/pubkey`, which means we would not be able to assume the file can be built on any machine. Adding `requiredSystemFeatures` to this derivation along with the derivations requiring the private keys felt more appropriate, because at the very least the public key can probably be safely assumed that it will exist on the same build machine that the private keys exist on?

This is also why I am setting `publicCertificateDer` as an environment variable that gets referenced in the build phase.

Reply:
I'd want to minimize the number of derivations that need to be built on a machine with requiredSystemFeatures, so I'd lean towards my approach.

Why use a environment variable instead of just substituting it in the nix string?

Reply:
Sounds good. I removed requiredSystemFeatures on the derivation just using the public key. We can probably add a check to that option field that uses `lib.isStorePath` or something similar in a future PR.